backtrace-library/src/main/java/backtraceio/library/metrics/BacktraceMetrics.java

```diff
@@ -0,0 +1,309 @@
+        return (ConcurrentLinkedDeque) this.summedEventsHandler.events;
+    }
+
+    /**
+     * Send startup event to Backtrace
+     */
+    public void sendStartupEvent() {
+        uniqueEventsHandler.sendStartupEvent(startupUniqueEventName);
+        summedEventsHandler.sendStartupEvent(startupSummedEventName);
+    }
+
+    /**
+     * Send all outgoing messages (unique and summed) currently queued
+     */
+    public void send() {
+        uniqueEventsHandler.send();
+        summedEventsHandler.send();
+    }
+
+    /**
+     * Add a unique event to the next Backtrace Metrics request
+     * @param attributeName Attribute name
+     * @return true if success
+     */
+    public boolean addUniqueEvent(String attributeName) {
+        return addUniqueEvent(attributeName, null);
+    }
+
+    /**
+     * Add a unique event to the next Backtrace Metrics request
+     * @param attributeName Attribute name
+     * @param attributes    Event attributes
+     * @return true if success
+     */
+    public boolean addUniqueEvent(String attributeName, Map<String, Object> attributes) {
+        if (!shouldProcessEvent(attributeName))
+        {
+            BacktraceLogger.w(LOG_TAG, "Skipping report: Reached store limit or event has empty name.");
+            return false;
+        }
+
+        Map<String, Object> localAttributes = new HashMap<String, Object>();
+
+        if (attributes != null) {
+            localAttributes.putAll(attributes);
+        }
+
+        BacktraceAttributes backtraceAttributes = new BacktraceAttributes(context, null, customAttributes);
+        localAttributes.putAll(backtraceAttributes.getAllBacktraceAttributes());
+
+        DeviceAttributesHelper deviceAttributesHelper = new DeviceAttributesHelper(context);
+        localAttributes.putAll(deviceAttributesHelper.getDeviceAttributes());
+
+        // validate if unique event attribute is available and
+        // prevent undefined attributes
+        if (localAttributes.get(attributeName) == null || localAttributes.get(attributeName).toString().trim().isEmpty())
+        {
+            BacktraceLogger.w(LOG_TAG, "Attribute name for Unique Event is not available in attribute scope");
+            return false;
+        }
+        // skip already defined unique events
+        for (Event event : uniqueEventsHandler.events) {
+            if (!(event instanceof UniqueEvent)) {
+                BacktraceLogger.e(LOG_TAG, "An event which is not a UniqueEvent found in UniqueEvents queue, this should not happen");
+                continue;
+            }
+            UniqueEvent uniqueEvent = (UniqueEvent) event;
+            if (uniqueEvent.getName().equals(attributeName)) {
+                BacktraceLogger.w(LOG_TAG, "Already defined unique event with this attribute name, skipping");
+                return false;
+            }
+        }
+
+        UniqueEvent uniqueEvent = new UniqueEvent(attributeName, System.currentTimeMillis() / 1000, localAttributes);
```

Comment:
`System.currentTimeMillis() / 1000`

internal functionality of the uniqueEvent class?

Reply:
Just in case I will add one more constructor so we never have to manually specify a timestamp if we don't want to @vlussenburg @konraddysput